Main.hs

```diff
@@ -90,16 +94,16 @@ basilica origin db emailChan = scottyApp $ do
       where len = Strict.length name
     postPost idParent = do
       content <- param "content"
-      maybeUser <- lmio (getUserByToken db) =<< getHeader "X-Token"
+      maybeUser <- lmdb db getUserByToken =<< getHeader "X-Token"
       case maybeUser of
         Nothing -> status status401 >> text "invalid token"
         Just user -> do
-          maybePost <- liftIO (createPost db user content idParent)
+          maybePost <- liftDB db (createPost user content idParent)
           maybe (post404 idParent) json maybePost
 
-withUser :: Database -> TokenRecord -> IO (TokenRecord, User)
-withUser db token@TokenRecord{..} = do
-  user <- fromJust <$> getUser db tokenUserID
```

Comment:
This is a nice, simple function to demonstrate the benefits of the `ReaderT` approach in isolation, and what the transformation process looks like. The `Database` argument goes away, the return type changes to `DatabaseM` instead of `IO`, but most importantly -- I no longer have to forward the `db` to the `getUser` call. Before I had to request it explicitly and forward it explicitly, and now that's both implicit due to the return type. It's neat!